StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1011ClosingSquareBracketsMustBeSpacedCorrectly.cs

```diff
@@ -117,7 +117,7 @@ private static void HandleCloseBracketToken(SyntaxTreeAnalysisContext context, S
                     break;
 
                 case SyntaxKind.QuestionToken:
-                    precedesSpecialCharacter = nextToken.Parent.IsKind(SyntaxKind.ConditionalAccessExpression);
+                    precedesSpecialCharacter = nextToken.Parent.IsKind(SyntaxKind.ConditionalAccessExpression) || nextToken.Parent.IsKind(SyntaxKind.NullableType);
```

Comment:
❓ Is this needed as part of the initial introduction of the C# 8 harness, or could it be submitted as a follow-up issue to keep this PR clean?

Reply:
I would have preferred to see #2927 fixed separately due to the size of the rest of the change, but if the CI passes I won't block this PR.

Reply:
In hindsight that would have been the better option, but given that the fix for #2927 is really small I didn't want to go through the hassle of separating them.